src/frontends/pytorch/src/op/max_poolnd.cpp

```diff
@@ -42,8 +54,55 @@ OutputVector translate_max_poolnd(const NodeContext& context) {
         rounding_type = context.const_input<bool>(5) ? RoundingType::CEIL : RoundingType::FLOOR;
     }
 
-    return {context.mark_node(
-        std::make_shared<v8::MaxPool>(context.get_input(0), strides, dilations, pads, pads, kernel, rounding_type))};
+    auto input = context.get_input(0);
+    if (rounding_type == RoundingType::CEIL) {
+        // The corner case of Max Pooling with ceil_mode on
+        // PyTorch allows sliding window go off bound, which leads to this accommodation.
+        // More detail on https://github.com/pytorch/pytorch/issues/57178
+        const auto zero = context.mark_node(v0::Constant::create(element::i32, Shape{}, {0}));
+        const auto one = context.mark_node(v0::Constant::create(element::i32, Shape{}, {1}));
+        const auto two = context.mark_node(v0::Constant::create(element::i32, Shape{}, {2}));
+
+        const auto padding =
+            context.input_is_none(3)
+                ? context.mark_node(std::make_shared<v0::Constant>(element::i32, Shape{pads.size()}, 0))
+                : context.get_input(3).get_node_shared_ptr();
```

Comment:
```suggestion
        const auto padding =
            context.input_is_none(3)
                ? context.mark_node(std::make_shared<v0::Constant>(element::i32, Shape{pads.size()}, 0))->output(0)
                : context.get_input(3);
```
It is not good to retrieve node from input. That node can have multiple outputs, but you will use only first output next, that may create incorrect graph.